includes/handler.inc

```diff
@@ -391,37 +435,36 @@ function islandora_oai_datetime_to_solr_time($datetime, $until = FALSE) {
  * @param string $query
  *   Query string to search for.
  * @param bool $set
- *   Whether we are querying down for a set which requires walking in ITQL land.
+ *   Whether we are recursively querying through a set.
  * @param bool $date_query
  *   Whether this is a date query or not.
  *
  * @return array
  *   The results generated from the crafted Solr query.
  */
 function islandora_oai_query_solr($start = 0, $field = 'PID', $query = NULL, $set = NULL, $date_query = NULL) {
-  global $user;
-
   $query_processor = new IslandoraSolrQueryProcessor();
 
   // Build the query string.
   if (empty($query)) {
     $query = '*:*';
   }
   else {
-    $query = $field . ':' . Apache_Solr_Service::escape($query);
+    module_load_include('inc', 'islandora_solr', 'includes/utilities');
+    $query = $field . ':' . islandora_solr_lesser_escape($query);
   }
-
   $query_processor->buildQuery($query);
 
+  // Build the fields to return.
   $fl_fields = islandora_oai_get_membership_array();
   $fl_fields[] = 'PID';
   $fl_fields[] = variable_get('islandora_oai_date_field', 'fgs_lastModifiedDate_dt');
   $query_processor->solrParams['fl'] = implode(',', $fl_fields);
 
+  // Build the filter array.
   if ($date_query) {
     $query_processor->solrParams['fq'][] = $date_query;
   }
-
   $exclude_content_models = islandora_oai_get_restricted_models();
   $has_model = variable_get('islandora_oai_content_model_field', 'RELS_EXT_hasModel_uri_ms');
```

Comment:
This one was existing so I'll leave it, but I've created https://jira.duraspace.org/browse/ISLANDORA-1783 to deal with deprecating the duplicate variable, and I'll use the `islandora_solr` version of the variable for the new ones I'm making.